pallets/pallet-bonded-coins/src/mock.rs

```diff
@@ -11,3 +46,295 @@ pub fn assert_relative_eq(target: Float, expected: Float, epsilon: Float) {
+		type BlockLength = ();
+		type BlockWeights = ();
+		type DbWeight = RocksDbWeight;
+		type Hash = Hash;
+		type Hashing = BlakeTwo256;
+		type Lookup = IdentityLookup<Self::AccountId>;
+		type MaxConsumers = ConstU32<16>;
+		type Nonce = u64;
+		type OnKilledAccount = ();
+		type OnNewAccount = ();
+		type OnSetCode = ();
+		type PalletInfo = PalletInfo;
+		type RuntimeCall = RuntimeCall;
+		type RuntimeEvent = RuntimeEvent;
+		type RuntimeOrigin = RuntimeOrigin;
+		type RuntimeTask = ();
+		type SS58Prefix = SS58Prefix;
+		type SystemWeightInfo = ();
+		type Version = ();
+	}
+
+	parameter_types! {
+		pub const ExistentialDeposit: Balance = 500;
+		pub const MaxLocks: u32 = 50;
+		pub const MaxReserves: u32 = 50;
+	}
+
+	impl pallet_balances::Config for Test {
+		type AccountStore = System;
+		type Balance = Balance;
+		type DustRemoval = ();
+		type ExistentialDeposit = ExistentialDeposit;
+		type FreezeIdentifier = ();
+		type MaxFreezes = ();
+		type MaxLocks = MaxLocks;
+		type MaxReserves = MaxReserves;
+		type ReserveIdentifier = [u8; 8];
+		type RuntimeEvent = RuntimeEvent;
+		type RuntimeFreezeReason = ();
+		type RuntimeHoldReason = RuntimeHoldReason;
+		type WeightInfo = ();
+	}
+
+	parameter_types! {
+		pub const StringLimit: u32 = 50;
+
+	}
+	impl pallet_assets::Config for Test {
+		type ApprovalDeposit = ConstU128<0>;
+		type AssetAccountDeposit = ConstU128<0>;
+		type AssetDeposit = ConstU128<0>;
+		type AssetId = AssetId;
+		type AssetIdParameter = AssetId;
+		type Balance = Balance;
+		type CallbackHandle = ();
+		type CreateOrigin = EnsureSigned<AccountId>;
+		type Currency = Balances;
+		type Extra = ();
+		type ForceOrigin = EnsureRoot<AccountId>;
+		type Freezer = ();
+		type MetadataDepositBase = ConstU128<0>;
+		type MetadataDepositPerByte = ConstU128<0>;
+		type RemoveItemsLimit = ConstU32<5>;
+		type RuntimeEvent = RuntimeEvent;
+		type StringLimit = StringLimit;
+		type WeightInfo = ();
+	}
+	parameter_types! {
+		pub const CurrencyDeposit: Balance = 500;
+		pub const MaxCurrencies: u32 = 50;
+		pub const CollateralAssetId: u32 = u32::MAX;
+	}
+
+	impl Config for Test {
+		type AssetId = AssetId;
+		type BaseDeposit = ExistentialDeposit;
+		type CollateralCurrency = Assets;
+		type CurveParameterInput = FloatInput;
+		type CurveParameterType = Float;
+		type DefaultOrigin = EnsureSigned<AccountId>;
+		type DepositCurrency = Balances;
+		type DepositPerCurrency = CurrencyDeposit;
+		type ForceOrigin = EnsureRoot<AccountId>;
+		type Fungibles = Assets;
+		type MaxCurrencies = MaxCurrencies;
+		type MaxStringLength = StringLimit;
+		type PoolCreateOrigin = EnsureSigned<AccountId>;
+		type PoolId = AccountId;
+		type RuntimeEvent = RuntimeEvent;
+		type RuntimeHoldReason = RuntimeHoldReason;
+	}
+
+	#[derive(Clone, Default)]
+	pub(crate) struct ExtBuilder {
+		native_assets: Vec<(AccountId, DepositCurrencyBalanceOf<Test>)>,
+		bonded_balance: Vec<(AssetId, AccountId, Balance)>,
+		// denomination, pool_id, PoolDetails
```

Comment:
what denomination is this that it is not part of the pool details?

Reply:
Oh, yes right. The denomination is also in the pool details presented. That makes it much easier. [e2ce89c](https://github.com/KILTprotocol/kilt-node/pull/780/commits/e2ce89cbb5fa2dac5efd5d8c5bf1dfee9568fee3)